### ISSUE TITLE
Add "Discard draft" option to Collection Edit page

### ISSUE THREAD

**Issue by amyehodge** [sul-dlss/happy-heron#794]
wireframes: https://projects.invisionapp.com/share/EQXC9CLKCR2#/screens/420930437
design doc: https://docs.google.com/document/d/1fcr2DYo7OrX-qTdeUOTWKSS1rlq_47sehESmbvgUsrk/edit#heading=h.z1sansuktydz

Update: Cancel button now in #857

**Comment by jcoyne** [comment]
@amyehodge the design doc describes this a "Discard draft" should we use that text or "Discard"?  Presumably this only appears if it is a draft and has already been saved because otherwise "Cancel" would have the same effect.

**Comment by amyehodge** [comment]
I've updated this to "Discard draft" as specified in the design doc.

And, yes, that makes sense that it's only there if it's an already saved draft.

**Comment by peetucket** [comment]
I am seeing the "Discard Draft" button on the Collection Edit page (I remember adding this last year).  It only appears for draft collections (not deposited).  So, is this about adding the cancel button, which goes back to the dashboard?

**Comment by peetucket** [comment]
![Screen Shot 2021-01-14 at 2.29.12 PM.png](https://images.zenhubusercontent.com/59b9a35fb0222d5de477a1fb/9ae58052-2adf-4d46-8d66-9cc790c549cd)

**Comment by amyehodge** [comment]
@peetucket This is a confusing topic. 

TL;DR
Both buttons are needed except in the case where the item has never been saved at all, i.e. user has clicked on "Create new collection" but never saved the form.

All the details:

The "Cancel" button
- cancels out of the data entry form without saving anything
-  1) if the item had never been saved, then nothing is saved here either, ie. there is no item listed in tables on the dashboard or elsewhere and no trace of it should remain.
-  2) if the item had previously been saved, then the item returns to the last saved state and none of the data that was entered into the form (if any) is saved. 
- in either case, returns the user to the collection view page 

- The "Discard draft" button
- deletes the entire draft, meaning 
- 1) if the item had never been deposited, then the draft will no longer appear on the dashboard
- 2) if the item had been deposited and this is a VERSION draft, then the item reverts back to the last version deposited, and is listed as "Deposited" in the tables. This includes removing any changes that had been made to the item when it was saved as a draft previously. 
- clicking this will bring up a pop up that asks “Are you sure you want to delete this draft? You cannot undo this action.”
- in either case, returns the user to the collection view page 

This leaves one overlapping scenario, which is that for items that have never been saved, that means they have also never been deposited, so for these "Cancel" button and the "Discard draft" button do the same thing, as Justin noted above. 

So, both buttons are needed except in the case where the item has never been saved at all, i.e. user has clicked on "Create new collection" but never saved the form.

(God, I hope I got this all straight.)

**Comment by peetucket** [comment]
Ok, I think I understand.  We currently have implemented "Discard Draft" for both Works and Collections, but it only works on the first draft, before deposit.  It just deletes the entire work or collection, making it disappear from the dashboard/listings.

Adding a cancel button (in all cases) that acts like a browser back button seems like it will fulfill the "cancel" requirements. Basically, back out of the form without saving changes to the previous state.  That is easy.

Allowing the discarding of previously saved second drafts of works and collections may be trickier - may need to consult with Mike and Justin about how this would work.

**Comment by jcoyne** [comment]
I think we all need to discuss about if the "revert to previously published version".  That is quite challenging, has particular problems with collections (where the set defaults may already be used by items in that collection), and I don't believe is something that Hydrus currently does.

**Comment by amyehodge** [comment]
Hydrus doesn't do this currently, and it makes things very confusing for users at times. This was an effort to correct this problem. 

I think we need to review which collection are changes taking effect when. As I'm thinking on this, it's complicated, because there are definitely changes that I don't think should be being used by collection items until the collection is redeposited.

**Comment by peetucket** [comment]
Update: taking cancel button to #857, updating this to reflect discard option only

**Comment by peetucket** [comment]
Also see #894 for works

**Comment by amyehodge** [comment]
Related to #1055, #1110, #1112

**Comment by amyehodge** [comment]
Just verifying that I see we have not yet addressed the use case with versions described above. That is, once the collection has been deposited, a user does not see a "discard draft" button and therefore can't revert to the previous saved version.

**Comment by jcoyne** [comment]
<img width="1189" alt="Screen Shot 2021-06-18 at 10 17 50 AM" src="https://user-images.githubusercontent.com/92044/122583053-737b7880-d01e-11eb-8717-f27fcdb8c22f.png">

@amyehodge I see a "discard draft" button on the "collection details" edit page after you save a draft of a new version. Is this ticket complete?

**Comment by amyehodge** [comment]
This does seem to be working now.
